amida-tech/amida-messaging-microservice
# Bug/SER-188 Fix Reference Message ID

#### What's this PR do?
Fixes the calculation of the referenceMessageId in thread summaries.
The value is the oldest created unread message in the thread if it exists, otherwise, the most recently created message in the thread.
This PR initializes the `maxCreatedDate` value to fix loop comparisons, and uses the `createdAt` value instead of the `readAt` value as definition of newest

#### Related JIRA tickets:
[SER-188](https://jira.amida-tech.com/browse/SER-188)

#### How should this be manually tested?
The case that this PR fixes is when all messages in a thread are read.
1. Create a thread with all read messages
1. Query for all threads
1. Check that the `refMessageId` for your thread is the most recently created message, not the oldest

#### Any background context you want to provide?
#### Screenshots (if appropriate):
